h/panels/navbar.py

```diff
@@ -29,14 +30,15 @@ def navbar(context, request, search=None, opts=None):
                 'title': group.name,
                 'link': request.route_url('group_read', pubid=group.pubid, slug=group.slug)
             })
-            groups_suggestions.append({
-                'name': group.name,
-                'pubid': group.pubid
-            })
         user_activity_url = request.route_url('activity.user_search',
                                               username=request.user.username)
         username = request.user.username
 
+    # Make all groups associated with the user visible in the search auto complete.
+    factory = list_groups_factory(None, request)
```

Comment:
Is there a reason for calling the factory directly here rather than using the `request.service("list_groups")` method, which ensures that only one instance of the service is created per request?